duplicates/src/192/main/kotlin/org/rust/ide/clones/RsDuplicateScope.kt

```diff
@@ -0,0 +1,176 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.clones
+
+import com.intellij.lang.LighterAST
+import com.intellij.lang.LighterASTNode
+import com.intellij.lang.LighterASTTokenNode
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectFileIndex
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.TokenType
+import com.intellij.psi.tree.TokenSet
+import com.jetbrains.clones.core.LighterAstNodeHashCache
+import com.jetbrains.clones.core.NodeHash
+import com.jetbrains.clones.core.longHash
+import com.jetbrains.clones.core.nodeListHash
+import com.jetbrains.clones.languagescope.common.DuplicateScopeBase
+import org.rust.lang.core.psi.RS_COMMENTS
+import org.rust.lang.core.psi.RS_LITERALS
+import org.rust.lang.core.psi.RsElementTypes.*
+import org.rust.lang.core.psi.tokenSetOf
+
+class RsDuplicateScope : DuplicateScopeBase() {
+
+    /**
+     * Presentable language name displayed in settings window
+     *
+     * Used as a DuplicateScope id in settings serialization
+     */
+    override val languageName: String = "Rust"
+
+    /**
+     * @return true if [file] should be analyzed
+     */
+    override fun acceptsFile(project: Project, file: VirtualFile): Boolean =
+        ProjectFileIndex.getInstance(project).isInSourceContent(file)
```

Comment:
let's check that the file is linked to a module structure, i.e. crate root exists.
Also maybe check that package origin is workspace instead of `isInSourceContent`?

Reply:
Unfortunately, we can't get crate root here because it is used while indexing and we get `IndexNotReadyException` because `RsFile.crateRoot` relies on our indices, so we can get rather restricted info in this place

Reply:
done